src/Analyzers/CSharp/Analyzers/NamespaceFileSync/NamespaceFileSyncDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,177 @@
+                .Count();
+            if (namespaceDeclCount > 1)
+            {
+                return false;
+            }
+
+            // It should not contain a namespace
+            var containsNamespace = namespaceDeclaration
+                 .DescendantNodes(n => n is NamespaceDeclarationSyntax)
+                 .OfType<NamespaceDeclarationSyntax>().Any();
+            if (containsNamespace)
+            {
+                return false;
+            }
+
+            // It should not contain partial classes with more than one instance in the semantic model
+            var containsPartialType = ContainsPartialTypeWithMultipleDeclarations(namespaceDeclaration, semanticModel);
+            if (containsPartialType)
+            {
+                return false;
+            }
+
+            // The current namespace should be valid
+            // TODO : can this cause a loop? This condition was used by the service before,
+            // but now could possibly add an Error severity as an analyzer
+            var isCurrentNamespaceInvalid = namespaceDeclaration.Name
+                .GetDiagnostics().Any(d => d.Severity == DiagnosticSeverity.Error);
+            if (isCurrentNamespaceInvalid)
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        private static bool ContainsPartialTypeWithMultipleDeclarations(NamespaceDeclarationSyntax namespaceDeclaration, SemanticModel semanticModel)
+        {
+            var partialMemberDecls = namespaceDeclaration
+                .Members.OfType<TypeDeclarationSyntax>()
+                .Where(t => t.Modifiers.Any(SyntaxKind.PartialKeyword));
+
+            foreach (var memberDecl in partialMemberDecls)
+            {
+                var memberSymbol = semanticModel.GetDeclaredSymbol(memberDecl);
+
+                // Simplify the check by assuming no multiple partial declarations in one document
+                if (memberSymbol is ITypeSymbol typeSymbol && typeSymbol.DeclaringSyntaxReferences.Length > 1)
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        private static bool IsFileAndNamespaceValid(
+            NamespaceDeclarationSyntax namespaceDeclaration,
+            string rootNamespace,
+            string projectDir,
+            [NotNullWhen(returnValue: true)] out string? targetNamespace)
+        {
+            var filePath = namespaceDeclaration.SyntaxTree.FilePath;
+            if (!filePath.Contains(projectDir))
+            {
+                // The file does not exist withing the project directory
+                targetNamespace = null;
+                return false;
+            }
+
+            var relativeFilePath = filePath.Substring(projectDir.Length);
+            var folders = Path.ChangeExtension(relativeFilePath, null).Split(Path.DirectorySeparatorChar).Where(e => e.Length > 0);
```

Comment:
Nit: Could pass in `StringSplitOptions.RemoveEmptyEntries` and avoid the `Where` method